compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java

```diff
@@ -9973,6 +9996,74 @@ private STNode parseNamespacePrefix() {
         }
     }
 
+    private STNode parseClientDeclOrVarDeclStatement(STNode metadata, List<STNode> qualifiers) {
+        return parseClientDeclOrVarDecl(metadata, STNodeFactory.createEmptyNode(), qualifiers, false);
+    }
+
+    private STNode parseClientDeclOrVarDecl(STNode metadata, STNode publicQualifier, List<STNode> qualifiers,
+                                            boolean isModuleVar) {
+        switch (peek().kind) {
+            case STRING_LITERAL_TOKEN:
+                reportInvalidMetaData(metadata, "client declaration");
+                reportInvalidQualifier(publicQualifier);
+                return parseClientDeclaration(qualifiers.get(qualifiers.size() - 1), isModuleVar);
+            case OBJECT_KEYWORD:
+                if (isModuleVar) {
+                    return parseModuleVarDecl(metadata, publicQualifier, qualifiers);
+                }
+                return parseVariableDecl(getAnnotations(metadata), publicQualifier, new ArrayList<>(), qualifiers,
+                                         false);
+            default:
+                recover(peek(), ParserRuleContext.CLIENT_DECL_OR_CLIENT_OBJECT_VAR_DECL);
```

Comment:
We can also have `CLASS_KEYWORD` after client keyword at module level

Reply:
We looked into this, right? And decided this was OK since class is covered in `parseTopLevelNode` https://github.com/ballerina-platform/ballerina-lang/blob/06c24debc25a7c923a0ae677e6ff461df34107dc/compiler/ballerina-parser/src/main/java/io/ballerina/compiler/internal/parser/BallerinaParser.java#L1019?

Reply:
Ah yes, btw there is an issue with this method. Created a separate issue to address later